### ISSUE TITLE
Any chance for Linux builds?

### ISSUE THREAD

**Issue by poVoq** [John3/Torque3D_Unofficial#3]
Would be great if there where also 64bit SDL Linux builds available.

Thanks!

**Comment by John3** [comment]
LOL yep I was thinking the same :D you got me just compiling T3D for linux right now :)

wait for the next week 👍 


![image](https://user-images.githubusercontent.com/1057449/35753043-c8d7e528-0823-11e8-8bdc-ecd619fa5b5c.png)

**Comment by John3** [comment]
Hi, I have compiled a Linux 64bit, using Ubuntu 16.04.3 LTS because is the only distro supported by Steam.
Any preferred distro you have in mind?

Only the build binary. You still need the template or module. Copy and paste as usual, if you are familiar with this repo.
http://www.mediafire.com/view/1qr6wm5f1appmkl (md5)
http://www.mediafire.com/view/bjf5wj4rgtijk94 (binary)

To decompress you can use the following command, or just right-click "extract"
`$ tar xJvf t3d_linux_x64.tar.xz `

**Comment by poVoq** [comment]
Ubuntu 16.04 should be ok for now and can probably be optimised to run on most other distributions. I'll test the builds later today and get back to you if I can get them to work (or not).

Thanks a lot!

**Comment by poVoq** [comment]
Menu starts up fine with the FPSGameplay module, but when trying to load a level or the world editor it crashes because it can't find the --/postFX/gl/postFX.glsl
This file is indeed not available in the basegame, and disabling all the advanced graphic settings also doesn't make a difference.

I also noticed that it still has the long standing Torque3D bug showing the audiodevice only as "null" and thus there is no sound. Somehow this was fixed in Terminal Overload a while ago.

Last but not least, the amount of crazy capitalisation in the asset files is a bit worry-some in a system that is case-sensitive. I can't see any immediate errors because of that, but I do wonder why that is still the case after having a Linux port for several years now :(

P.S.: My system is Kubuntu 16.04 with a Nvidia GPU.

**Comment by John3** [comment]
Hey thanks for the feedback, keep in mind that I'm using the last commit that have include new stuff like entity component and AFX, and this touch the templates.

> Menu starts up fine with the FPSGameplay module, but when trying to load a level or the world editor it crashes because it can't find the --/postFX/gl/postFX.glsl
> This file is indeed not available in the basegame, and disabling all the advanced graphic settings also doesn't make a difference.

Something was missed, may be something related to the last commit, pls try this templates. You need to copy the binary inside of:
Full/game/
And
Basegame/game/

http://www.mediafire.com/view/zkpzau28odvihfi
272549301d53329136acbc1daffd2741 *CLEAN_templates.7z

http://www.mediafire.com/view/v71w24v1l081amg
0b4836b902b26398da940298fc586985 *CLEAN_templates.tar.xz

> I also noticed that it still has the long standing Torque3D bug showing the audiodevice only as "null" and thus there is no sound.

Thx I was not aware. I need to check it.

> The amount of crazy capitalisation in the asset files is a bit worry-some in a system that is case-sensitive.
Yes sorry about that, this is something that will fixed after the big changes, because it will touch a lot of files.

Pls share your console.log file, and some pictures. I will have a fix more quickly.

**Comment by poVoq** [comment]
Ok maybe I am a bit stupid, but that really does not make a difference. With the Basegame the postFX.glsl ist still missing, here is a full console log until it crashes: https://pastebin.com/susv53Ce

And the fullgame... I have no idea how to use. Does it actually start into the engine? If I launch it with the provided binary it just goes into some engine prompt? Console log: https://pastebin.com/ArguiTMU

Thanks for the help, really apprechiated. And sorry again for being maybe a bit stupid about it.

**Comment by John3** [comment]
Don't worry, something is wrong with the latest commits.

In the meantime, please download this version with his full template. (T3D v3.10). Pretty sure this would work for you.
https://github.com/John3/Torque3D_Unofficial/releases/tag/v1.18

I will keep this issue open until I have the latest dev build.

Thank you for you feedback.

**Comment by poVoq** [comment]
Yes, except for the "null" device audio bug, this older build seems to work.

**Comment by John3** [comment]
Hi, I have the last commit ready to upload, I will take a few day to proper test this. Any feedback on your side? BTW the null audio device is not a bug, you have something that refer as bug?

**Comment by poVoq** [comment]
Ok, the null device issue is a dynamic linking problem under Ubuntu 16.04. Unless you install the libopenal1-dev package the sound doesn't work.

Otherwise, what kind of feedback can I give without a new build? I will gladly test your new builds :)

**Comment by John3** [comment]
> Ok, the null device issue is a dynamic linking problem under Ubuntu 16.04. Unless you install the libopenal1-dev package

thanks, what you remember, for what version of t3d did you see this?

> the sound doesn't work.

pls elaborate, it is not supposed to sound.

> what kind of feedback can I give without a new build?

Nothing in particular, if you forgot something 😆 

I appreciate your help, I think I'll finish sooner :)

**Comment by poVoq** [comment]
That is with the T3D v3.10 build you linked before (but I also had the same issue lot of times with any T3D based games... I guess most solved it by including a libopenal.so in their game instead of dynamically linking it?).

But that it only finds a "null" audio device also happened with the newer 4.0 builds that you initially shared when I tried them previously, of course there I couldn't actually hear any sound as it always crashed before entering the game.

**Comment by John3** [comment]
Hi @poVoq sorry for the delay, can you please try this release https://github.com/John3/Torque3D_Unofficial/releases/edit/v1.23 and use the Full template, decompress t3d_linux_x64.tar.xz and replace the game folder.

**Comment by poVoq** [comment]
Switched to Solus Linux in the mean time. Also there I had to install the OpenAL-devel files for sound to work. Otherwise everything seems to be working fine with the above build... except: is that test level with the road supposed to have that much bloom on highest settings? It looks a bit over-done.

**Comment by John3** [comment]
> Otherwise everything seems to be working fine with the above build

Great news poVoq!

> except: is that test level with the road supposed to have that much bloom on highest settings?

Yes at the moment afraid to said, you can adjusted in the PostFX menu nevertheless. Until development go toward to v4 this will go to change.

Ok now I will standardize my development environment to Linux, and start releasing this builds.

I need to release one more build, a recent one, to close this issue, this will have more integrate the entity component stuff that are rolling in the past weeks.

Can you help to give another try with this recent build?

**Comment by poVoq** [comment]
Sure, I'll test it when you give me a note here.

**Comment by John3** [comment]
The last devhead (up to date) build is ready to download from here: [v1.24](https://github.com/John3/Torque3D_Unofficial/releases/tag/v1.24) 

You are using Solus Linux v3? The only additional steps is install OpenAL-devel?

Thank you for your help.

**Comment by poVoq** [comment]
This one doesn't start, see console log here:
https://pastebin.com/BZgPePi3

> You are using Solus Linux v3? The only additional steps is install OpenAL-devel?

With 1.23 yes.

**Comment by John3** [comment]
can you copy the main.cs from the "Full" template folder and replace the one you have in your folder?

Or delete /home/kris/Dev/Games/TO_reboot/engine/linux_builds/Full/main.cs and paste the one that have the Full template.

I think you have the wrong main.cs. 

EDIT: I include a second main.cs pretty sure this was the error. Update the executable builds.

**Comment by poVoq** [comment]
Now it seems to work. Is there an easy way to test the bullet physics?

**Comment by John3** [comment]
> Now it seems to work. Is there an easy way to test the bullet physics?

Yes, just copy the bullet executable like you did before. Do this bullet test in the Full template. Remember the shape need to have collision. 

You can create the collision from inside Torque3D or from you 3D modelling software like Blender.

See one on my videos: https://youtu.be/3xTLFtU0ijQ?t=44s

**Comment by poVoq** [comment]
Trying to test the T3D Demos (Mars, Sector etc. from your other repository), but these seem to require the new basegame + FPStemplate and that crash with the same error as before e.g. --/postFX/gl/postFX.glsl

Also despite having copied the level files, they do not show up in the menu.
